### ISSUE TITLE
unique constraint seems not to work

### ISSUE THREAD

**Issue by stratboy** [balderdashy/sails-mysql#259]
Hi, this is a test model:

``` javascript
module.exports = {
  attributes: {
    username:{
      type:'string'
    }
    ,email:{
      type:'string'
      ,email:true
      ,unique: true
    }
  }
};
```
by lifting, sails complains:

Error (E_UNKNOWN) :: Encountered an unexpected error
: ER_TOO_LONG_KEY: Specified key was too long; max key length is 767 bytes

The row that cause this throw is the one that adds the unique constrain.

Maybe it's related to a waterline issue:

"There is currently an issue with adding indexes to string fields. Because Waterline performs its queries in a case insensitive manner, we are unable to use the index on a string attribute. There are some workarounds being discussed but nothing is implemented so far. This will be updated in the near future to fully support indexes on strings."

https://github.com/balderdashy/waterline

**Comment by sailsbot** [comment]
Thanks for posting, @stratboy.  I'm a repo bot-- nice to meet you!

It has been 30 days since there have been any updates or new comments on this page.  If this issue has been resolved, feel free to disregard the rest of this message.  On the other hand, if you are still waiting on a patch, please:

  + review our [contribution guide](https://github.com/balderdashy/sails/blob/master/CONTRIBUTING.md) to make sure this submission meets our criteria (only _verified bugs_ with documented features, please;  no questions, commentary, or bug reports about undocumented features or unofficial plugins)
  + create a new issue with the latest information, including updated version details with error messages, failing tests, and a link back to [the original issue]().  This allows GitHub to automatically create a back-reference for future visitors arriving from search engines.

Thanks so much for your help!

**Comment by LeBlaaanc** [comment]
Anyone looking for a quick solution for this. If you are running utf8mb4 be sure any attribute set to `unique: true` with `type: string/email` has `size:191`. Happy coding!
